datapipelines/src/test/java/io/customer/datapipelines/sdk/CustomerIOBuilderTest.kt

```diff
@@ -36,53 +35,57 @@ class CustomerIOBuilderTest : BaseUnitTest() {
         super.teardown()
     }
 
+    private fun mockGenericModule(): CustomerIOGenericModule {
+        return mockk<CustomerIOGenericModule>(relaxUnitFun = true)
+    }
+
     @Test
     fun build_givenModule_expectInitializeModule() {
-        val givenModule: CustomerIOGenericModule = mock<CustomerIOGenericModule>().apply {
-            whenever(this.moduleName).thenReturn(String.random)
+        val givenModule: CustomerIOGenericModule = mockGenericModule().apply {
+            every { moduleName } returns String.random
         }
 
         createCustomerIOBuilder()
             .addCustomerIOModule(givenModule)
             .build()
 
-        verify(givenModule).initialize()
+        verify(exactly = 1) { givenModule.initialize() }
     }
 
     @Test
     fun build_givenMultipleModules_expectInitializeAllModules() {
-        val givenModule1: CustomerIOGenericModule = mock<CustomerIOGenericModule>().apply {
-            whenever(this.moduleName).thenReturn(String.random)
+        val givenModule1: CustomerIOGenericModule = mockGenericModule().apply {
+            every { moduleName } returns String.random
         }
-        val givenModule2: CustomerIOGenericModule = mock<CustomerIOGenericModule>().apply {
-            whenever(this.moduleName).thenReturn(String.random)
+        val givenModule2: CustomerIOGenericModule = mockGenericModule().apply {
+            every { moduleName } returns String.random
         }
 
         createCustomerIOBuilder()
             .addCustomerIOModule(givenModule1)
             .addCustomerIOModule(givenModule2)
             .build()
 
-        verify(givenModule1).initialize()
-        verify(givenModule2).initialize()
+        verify(exactly = 1) { givenModule1.initialize() }
+        verify(exactly = 1) { givenModule2.initialize() }
```

Comment:
```suggestion
        verifyCount {
            1 * { givenModule1.initialize() } // same as verify(exactly = 1) {  }
            1 * { givenModule2.initialize() } // same as verify(exactly = 1) { }
        }
```

Reply:
This does sound good, but unfortunately, it requires newer version of Kotlin as it gives the following error:

> /Users/mrehan/.gradle/caches/transforms-3/b2c3058d9c2ad25669f5c8e3f7c7515a/transformed/jetified-kotlin-stdlib-common-1.9.10.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.9.0, expected version is 1.7.1.

Merging this for now. We can update this later when we upgrade the Kotlin version.